### ISSUE TITLE
IPython 5.0 (forthcoming) seems to have removed an API the line_profiler extension needs

### ISSUE THREAD

**Issue by njsmith** [rkern/line_profiler#62]
FYI - using a current (2016-05-30) version of IPython master (i.e. what will soon be released as 5.0), I get:

```
In [1]: %load_ext line_profiler
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-f2048e70c30c> in <module>()
----> 1 get_ipython().magic('load_ext line_profiler')

/home/njs/.user-python3.5-64bit/lib/python3.5/site-packages/IPython/core/interactiveshell.py in magic(self, arg_s)
   2151         magic_name, _, magic_arg_s = arg_s.partition(' ')
   2152         magic_name = magic_name.lstrip(prefilter.ESC_MAGIC)
-> 2153         return self.run_line_magic(magic_name, magic_arg_s)
   2154 
   2155     #-------------------------------------------------------------------------

/home/njs/.user-python3.5-64bit/lib/python3.5/site-packages/IPython/core/interactiveshell.py in run_line_magic(self, magic_name, line)
   2072                 kwargs['local_ns'] = sys._getframe(stack_depth).f_locals
   2073             with self.builtin_trap:
-> 2074                 result = fn(*args,**kwargs)
   2075             return result
   2076 

<decorator-gen-62> in load_ext(self, module_str)

/home/njs/.user-python3.5-64bit/lib/python3.5/site-packages/IPython/core/magic.py in <lambda>(f, *a, **k)
    186     # but it's overkill for just that one bit of state.
    187     def magic_deco(arg):
--> 188         call = lambda f, *a, **k: f(*a, **k)
    189 
    190         if callable(arg):

/home/njs/.user-python3.5-64bit/lib/python3.5/site-packages/IPython/core/magics/extension.py in load_ext(self, module_str)
     35         if not module_str:
     36             raise UsageError('Missing module name.')
---> 37         res = self.shell.extension_manager.load_extension(module_str)
     38 
     39         if res == 'already loaded':

/home/njs/.user-python3.5-64bit/lib/python3.5/site-packages/IPython/core/extensions.py in load_extension(self, module_str)
     83                     __import__(module_str)
     84             mod = sys.modules[module_str]
---> 85             if self._call_load_ipython_extension(mod):
     86                 self.loaded.add(module_str)
     87             else:

/home/njs/.user-python3.5-64bit/lib/python3.5/site-packages/IPython/core/extensions.py in _call_load_ipython_extension(self, mod)
    130     def _call_load_ipython_extension(self, mod):
    131         if hasattr(mod, 'load_ipython_extension'):
--> 132             mod.load_ipython_extension(self.shell)
    133             return True
    134 

/home/njs/.user-python3.5-64bit/lib/python3.5/site-packages/line_profiler.py in load_ipython_extension(ip)
    370     """ API for IPython to recognize this module as an IPython extension.
    371     """
--> 372     ip.define_magic('lprun', magic_lprun)
    373 
    374 

AttributeError: 'TerminalInteractiveShell' object has no attribute 'define_magic'
```

**Comment by caethan** [comment]
I have a pull request in to fix this:  https://github.com/rkern/line_profiler/pull/65

Please try it out and see if it works with your system.

**Comment by caethan** [comment]
Fixed in 2.0 release.
